packages/core/src/sync-realtime/service.ts

```diff
@@ -773,9 +782,19 @@ export class RealtimeSyncService extends Emittery<RealtimeSyncEvents> {
         method: "eth_getBlockByNumber",
         params: [typeof block === "number" ? numberToHex(block) : block, true],
       })
-      .then((block) => {
-        if (!block) throw new BlockNotFoundError({});
-        return block as RealtimeBlock;
+      .then((_block) => {
+        if (!_block)
+          throw new BlockNotFoundError(
```

Comment:
We could probably just pass `blockNumber: "latest"` here because it's only user in a string interpolation?
```ts
import type { Hash } from '../types/misc.js'

import { BaseError } from './base.js'

export type BlockNotFoundErrorType = BlockNotFoundError & {
  name: 'BlockNotFoundError'
}
export class BlockNotFoundError extends BaseError {
  override name = 'BlockNotFoundError'
  constructor({
    blockHash,
    blockNumber,
  }: {
    blockHash?: Hash
    blockNumber?: bigint
  }) {
    let identifier = 'Block'
    if (blockHash) identifier = `Block at hash "${blockHash}"`
    if (blockNumber) identifier = `Block at number "${blockNumber}"`
    super(`${identifier} could not be found.`)
  }
}
```